src/cursor/cursor.ts

```diff
@@ -55,6 +57,83 @@ export interface CursorOptions extends CoreCursorOptions {
   numberOfRetries?: number;
 }
 
+export class CursorStream extends Readable {
+  [kCursor]: Cursor;
+
+  /** @event */
+  static readonly CLOSE = 'close' as const;
+  /** @event */
+  static readonly DATA = 'data' as const;
+  /** @event */
+  static readonly END = 'end' as const;
+  /** @event */
+  static readonly FINISH = 'finish' as const;
+  /** @event */
+  static readonly ERROR = 'error' as const;
+  /** @event */
+  static readonly PAUSE = 'pause' as const;
+  /** @event */
+  static readonly READABLE = 'readable' as const;
+  /** @event */
+  static readonly RESUME = 'resume' as const;
+
+  constructor(cursor: Cursor) {
+    super({ objectMode: true });
+    this[kCursor] = cursor;
+    this.on('end', () => this[kCursor].close(() => this.emit(Cursor.CLOSE)));
+  }
+
+  destroy(err?: AnyError): void {
+    if (err) this.emit(Cursor.ERROR, err);
+    this.pause();
+    this[kCursor].close(() => this.emit(Cursor.CLOSE));
+  }
+
+  /** @internal */
+  _read(): void {
+    const cursor = this[kCursor];
+    if ((cursor.s && cursor.s.state === CursorState.CLOSED) || cursor.isDead()) {
+      this.push(null);
+      return;
+    }
+
+    // Get the next item
+    cursor._next((err, result) => {
+      if (err) {
+        if (this.listeners(CursorStream.ERROR) && this.listeners(CursorStream.ERROR).length > 0) {
+          this.emit(CursorStream.ERROR, err);
+        }
+        if (!cursor.isDead()) cursor.close();
+
+        // Emit end event
+        this.emit(CursorStream.END);
+        this.emit(CursorStream.FINISH);
+        return;
+      }
+
+      // If we provided a transformation method
+      if (
+        cursor.cursorState.streamOptions &&
+        typeof cursor.cursorState.streamOptions.transform === 'function' &&
+        result != null
+      ) {
+        this.push(cursor.cursorState.streamOptions.transform(result));
+        return;
+      }
+
+      // Return the result
+      this.push(result);
+
+      if (result === null && cursor.isDead()) {
+        this.once(CursorStream.END, () => {
+          cursor.close();
+          this.emit(CursorStream.FINISH);
+        });
+      }
```

Comment:
I'm again unclear whether we should be doing all of this anymore. A few questions:
- when `result === null`, I believe `cursor.isClosed() === true` can you confirm? If so, we're needlessly duplicating the `cursor.close()` here.
- if `result === null` then we `push(null)` and I believe that will cause the stream to end, so we're emitting duplicate events here

Reply:
@emadum did you get a chance to see this comment yet? I think it still stands.

Reply:
I tried removing the `cursor.close`, but a test ("should invalidate change stream on collection rename using event listeners") started timing out (even though `cursor.isClosed() === true` before calling `cursor.close()`). I can take a closer look.

Reply:
ok let's leave it alone for now